src/test/java/com/philips/research/bombar/core/domain/ProjectInteractorTest.java

```diff
@@ -210,6 +210,30 @@ void readsLicenseDistributionForProject() {
             assertThat(distribution).containsEntry("B", 1);
         }
 
+        @Test
+        void noObligationsForEmptyProject() {
+            project.clearDependencies();
+            final var ObligationsMap = interactor.getObligations(PROJECT_ID);
+            assertThat(ObligationsMap).isEmpty();
+        }
+
+        @Test
+        void obligationsInAProject() {
+            ProjectService mockInteractor = spy(interactor);
+            when(store.getProject(PROJECT_ID)).thenReturn(Optional.of(project));
+            final var dependencies = Set.of(dependency.setLicense("LICENSE"));
+            final var obligations = new HashMap<String, Set<Dependency>>();
+            obligations.put("Obligation", dependencies);
+            final var dependencyDto = DtoConverter.toBaseDto(dependency);
+
+            doReturn(obligationAnalyzer).when(mockInteractor).getObligationAnalyzerInstance(PROJECT_ID);
```

Comment:
The code declares a class member for the mock analyzer, but this is the only location where that object is used.

Consider moving creation of this mock into this method (and simply name it `analyzer`).